### PR TITLE
samples: Fix warning for minimal pinout FEM in DTM and radio_test

### PR DESCRIPTION
**Pull request by dchat-nordic**
Fix warning when DTM or radio_test is compiled with FEM in minimal pinout configuration.

### CODE REVIEW

File: samples/bluetooth/direct_test_mode/src/fem/generic_fem.c [hunk 1]
```diff
@@ -86,6 +86,14 @@ static int gpio_config(void)
 	}
 #endif /* DT_NODE_HAS_PROP(DT_NODELABEL(nrf_radio_fem), chl_gpios) */
 
+#ifdef __cplusplus
```
Comment (KAGA164) [inline samples/bluetooth/direct_test_mode/src/fem/generic_fem.c:89]: Wouldn't be a simpler solution to add the __used attribute to this structure definition?

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -171,6 +171,10 @@ Bluetooth samples
 
   * The Bluetooth 3-wire coex sample because of the removal of the 3-wire implementation.
 
+* :ref:`direct_test_mode` sample:
+
+  * Fixed a compilation warning when used with minimal pinout FEM.
```
Comment (KAGA164) [inline doc/nrf/releases/release-notes-changelog.rst:176]: Please mark that this is for generic or Skyworks fems

File: samples/bluetooth/direct_test_mode/src/fem/generic_fem.c (hunk 1 above)
Comment (dchat-nordic) [reply samples/bluetooth/direct_test_mode/src/fem/generic_fem.c:89]: Actually it would be simpler

File: doc/nrf/known_issues.rst [hunk 3]
```diff
@@ -3082,6 +3082,11 @@ NCSDK-19858: :ref:`at_monitor_readme` library and :ref:`nrf_cloud_mqtt_multi_ser
   Occasionally, the :ref:`at_monitor_readme` library heap becomes overrun, presumably due to one of the registered AT event listeners becoming stalled.
   This has only been observed with the :ref:`nrf_cloud_mqtt_multi_service` sample.
 
+.. rst-class:: v2-3-0 v2-2-0
+
+NCSDK-20095: Build warning in the RF test samples when the minimal pinout FEM is used
+  :ref:`radio_test` and :ref:`direct_test_mode` builds with warning for with the FEM in minimal pinout configuration.
```
Comment (peknis) [inline doc/nrf/known_issues.rst:3088]:
```suggestion
  The :ref:`radio_test` and :ref:`direct_test_mode` samples build with a warning about FEM in minimal pinout configuration.
```
Any workaround?

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -171,6 +171,10 @@ Bluetooth samples
 
   * The Bluetooth 3-wire coex sample because of the removal of the 3-wire implementation.
 
+* :ref:`direct_test_mode` sample:
+
+  * Fixed a compilation warning when used with minimal pinout Skyworks FEM.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:176]: This is a bit vague. Is the warning different after this fix or is it not shown anymore?

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst (hunk 4 above)
Comment (dchat-nordic) [reply doc/nrf/releases/release-notes-changelog.rst:176]: It is not shown anymore

**Review by peknis — APPROVED** [review]

**Review by maje-emb — COMMENTED** [review]
The changes involve the DTM sample, what about Radio Test? 
Based on the description, it seems that a warning also occurs for Radio Test.

**Comment by dchat-nordic** [comment]
> The changes involve the DTM sample, what about Radio Test? Based on the description, it seems that a warning also occurs for Radio Test.

The changed file is used both by Radio Test and DTM.

**Review by maje-emb — APPROVED** [review]

**Review by KAGA164 — APPROVED** [review]
LGTM

**Pull request merged**